gpt_index/readers/file.py

```diff
@@ -50,7 +65,9 @@ class SimpleDirectoryReader(BaseReader):
             Default is None.
         num_files_limit (Optional[int]): Maximum number of files to read.
             Default is None.
-
+        file_metadata (Optional[Callable]): A function that takes in a 
```

Comment:
would be good to define a more specific type, e.g. `Callable[str, Dict]`

Reply:
Done

Reply:
Shall we do the same with `file_extractor`?

Reply:
yeah sure! 
- if you don't want to deal with too many nested types you could define a variable e.g. `FILE_EXTRACTOR_FN_TYPE = Callable[[Path, bool], str]`
- I realized that wasn't even in the docstring, if you could add that that would be amazing 🙏